Source/Core/Core/HW/EXI/BBA/BuiltIn.cpp

```diff
@@ -0,0 +1,656 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Core/HW/EXI/BBA/BuiltIn.h"
+
+#include <cstring>
+
+#include <SFML/Network.hpp>
+
+#include "Common/Assert.h"
+#include "Common/Logging/Log.h"
+#include "Common/MsgHandler.h"
+#include "Common/StringUtil.h"
+#include "Common/Swap.h"
+#include "Core/HW/EXI/EXI_Device.h"
+#include "Core/HW/EXI/EXI_DeviceEthernet.h"
+#include "VideoCommon/OnScreenDisplay.h"
+
+// #define BBA_TRACK_PAGE_PTRS
+
+namespace ExpansionInterface
+{
+u64 GetTickCountStd()
+{
+  using namespace std::chrono;
+  return duration_cast<milliseconds>(steady_clock::now().time_since_epoch()).count();
+}
+
+bool CEXIETHERNET::BuiltInBBAInterface::Activate()
+{
+  if (IsActivated())
+    return true;
+
+  active = true;
+  m_in_frame = new u8[9004];
```

Comment:
in general, dolphin never uses bare `new` or `delete`, instead use `std::vector` or something from the <memory> header like `std::unique_ptr` etc